composer/loggers/tensorboard_logger.py

```diff
@@ -115,11 +131,32 @@ def _flush(self, logger: Logger):
         assert self.writer.file_writer is not None
         file_path = self.writer.file_writer.event_writer._file_name
 
+        # If no writes have happened since the last flush, then file_path won't exist, so
+        # we should skip doing flushing and skip filing the artifact since it will error
+        # out anyway (given that the file_path doesn't exist).
+        if not Path(file_path).exists():
+            return
+
+        assert self.writer is not None
+        self.writer.flush()
+
+
         logger.file_artifact(
             LogLevel.FIT,
             # For a file to be readable by Tensorboard, it must start with
             # 'events.out.tfevents'. Child directory is named after run_name, so the logs
             # are named properly in the Tensorboard GUI.
-            artifact_name='tensorboard_logs/{run_name}/events.out.tfevents-{run_name}-{rank}',
+            artifact_name=(
+                'tensorboard_logs/{run_name}/events.out.tfevents-{run_name}-{rank}'
+                # Append flush_count, so artifact has unique name.
+                + f'-{self.flush_count}'),
             file_path=file_path,
             overwrite=True)
+        
+        # Close writer and reinitialize it with a new log file path. This ensures that 
+        # we have one file per flush, which means `file_artifact` is always copying 
+        # data points that have never been copied before. Also this ensures no strange 
+        # issues with dropping of points when setting a new log file path.
+        self.writer.close()
+        self.flush_count += 1
+        self._initialize_summary_writer()
```

Comment:
Seems convoluted to me that with every call to `flush()`, we are basically closing the writer and creating a new one (to a new file). I don't think this complicated handling is really needed, but rather just set a higher `flush_interval`?

Reply:
Thanks for the explanation, I understand the need. However, there may be a simpler way here to have multiple event files, given that `SummaryWriter` will automatically re-open new `EventWriter`, which creates a new event file inside the logdir, e.g.
```
writer = SummaryWriter(log_dir='test_tb')
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.close()
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.close()
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.add_scalar(tag='hello', scalar_value=5)
writer.close()
```

will populate the `log_dir` with:
```
events.out.tfevents.1657784030.Hanlins-M1-Macbook-Pro.local.216.0
events.out.tfevents.1657784030.Hanlins-M1-Macbook-Pro.local.216.1
events.out.tfevents.1657784030.Hanlins-M1-Macbook-Pro.local.216.2
```

Is that sufficient to implement this behavior?

Reply:
Ooh interesting, if `.close()` does that for us, then yes, that would be way simpler.

Reply:
Yup it's handled by the `SummaryWriter`, and the tensorbord event writer ensures the files are unique with a guid, this might simplify the implementation significantly.

Reply:
Great find, Hanlin! Changed it to use close. Greatly simplified a lot of logic. thanks. fixed